[Docs] Add maximumRetryAttempts to AWS guide

rdarcy1 opened the pull request:
Document max retry attempts for async Lambda invocations: https://docs.aws.amazon.com/lambda/latest/dg/invocation-async.html

It's supported by Serverless, just not documented as far as I can see

pgrzesik commented on docs/providers/aws/guide/serverless.yml.md:
```diff
@@ -132,6 +132,8 @@ provider:
   # Processor architecture: 'x86_64' or 'arm64' via Graviton2 (default: x86_64)
   architecture: x86_64
+  # Maximum retry attempts when an asynchronous invocation fails (between 0 and 2; default: 2)
+  maximumRetryAttempts: 1
```
Actually this is not supported on `provider` level - it has to be set directly on a function and it's documented in `functions.md` file. It would be good to add it in this fine, but in `Functions` section.

pgrzesik reviewed (changes requested):
Thanks @rdarcy1 - good catch, but the placement of that property is a bit different - could you take a look at my comment please?

rdarcy1 commented:
My bad, moved it to the function block so thing it's correct now? Also added max event age as they're documented together in functions.md (missed that bit of documentation!)

pgrzesik commented on docs/providers/aws/guide/serverless.yml.md:
```diff
@@ -676,6 +676,10 @@ functions:
       localMountPath: /mnt/example
+    # Maximum retry attempts when an asynchronous invocation fails (between 0 and 2; default: 2)
+    maximumRetryAttempts: 1
+    # Maximum event age in seconds when invoking asynchornously (between 60 and 21600)
```
there's a small typo - `asynchornously` -> `asynchronously`

pgrzesik reviewed (dismissed):
thanks @rdarcy1 - that looks about right 👍 There's a minor typo - could you take a look at my comment? After that we're good to go 👍
